ssl/s3_lib.c

```diff
@@ -3770,6 +3769,34 @@ OPENSSL_GLOBAL const SSL_CIPHER ssl3_ciphers[] = {
      256,
      },
 
+    {
+     1,
+     "GOST2012-GOST8912-GOST8912",
+     0x0300ff85,
+     SSL_kGOST,
+     SSL_aGOST12 | SSL_aGOST01,
+     SSL_eGOST2814789CNT12,
+     SSL_GOST89MAC12,
+     SSL_TLSV1,
+     SSL_NOT_EXP | SSL_HIGH,
+     SSL_HANDSHAKE_MAC_GOST12_256 | TLS1_PRF_GOST12_256 | TLS1_STREAM_MAC,
```

Comment:
After backporting the fix you have suggested to 1.2, I get an error on new ciphersuites:

140033947571856:error:1408C095:SSL routines:ssl3_get_finished:digest check failed:s3_both.c:273:
140033947571856:error:140780E5:SSL routines:ssl23_read:ssl handshake failure:s23_lib.c:137:

I still do not remember exactly, why did I introduce a separate function. I will study it again.